pyfftw/interfaces/_utils.py

```diff
@@ -97,24 +101,25 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
         # take the transform of the input array! (in general, we have
         # to assume that the input array will be destroyed during 
         # planning).
-        a_copy = a.copy()
+        if overwrite_input:    # Note: reload_after_transform is not True...
+            args = (a, s, axes) + args
+            a_copy = a.copy()  # ... hence this is not an additional copy
+        else:
+            args = (numpy.empty_like(a), s, axes) + args
```

Comment:
Ah, I was slightly wrong. [`FFTW.__call__`](https://github.com/mforbes/pyFFTW/blob/patch-1/pyfftw/pyfftw.pyx#L1339) will force the array to be correct before calling `update_arrays`, which means an extra copy. The problem is there, but is a performance related one as you pointed out.